doc/screen-recording.md

```diff
@@ -0,0 +1,69 @@
+# Genesys Cloud WebRTC SDK Screen Recording
+
+This SDK supports abstracting the screen recording process that occurs inside the Genesys Cloud Desktop App. Screen 
+recordings are initiated by the server which notifies the webrtc sdk who then abstracts nearly all of the setup and
+signaling to make the screen recording work.
+
+> *Note: screen recording does not support guest users.*
+
+## Prerequisites
+
+You must have screen recording policies in place and the Webrtc Sdk must be configured to allow screen recording sessions.
+
+## Example Usage
+Once you have an initialized instance of the WebrtcSdk, you can join or create a conference in the following manner:
+
+### Automatic Accept (default)
+Signaling is automatically accepted by the sdk. When the screen recording session comes in, all you need to do is add
+your screen tracks and then call `sdk.acceptSession(session)`.
+
+``` ts
+// set up needed events
+sdk.on('sessionStart', async (session) => {
+  if (session.sessionType === SessionTypes.screenRecording) {
+    // gather media in whatever way you want. The SDK does *not* gather screen media for you
+    const screenStream = await navigator.getDisplayMedia();
+
+    // add the media track(s) to the session
+    (session as ).addScreenTrackToSession()
+    sdk.acceptSession({ conversationId: session.conversationId, audioElement, videoElement });
+  }
+});
+
+await sdk.startVideoConference(roomJid);
+```
+
+### Manual Accept
+Signaling is automatically accepted by the sdk. When the screen recording session comes in, all you need to do is add
+your screen tracks and then call `sdk.acceptSession(session)`.
+
+``` ts
+const sdk = new GenesysCloudWebrtcSdk({
+  // other config stuff ...
+  autoAcceptScreenRecordingRequests: true
+});
+
+...
+
+sdk.on('pendingSession', (session) => {
+  if (pendingSession.sessionType === SessionTypes.screenRecording) {
+    sdk.acceptPendingSession()
+  }
+});
+
+// set up needed events
+sdk.on('sessionStart', async (session) => {
+  if (session.sessionType === SessionTypes.screenRecording) {
+    // gather media in whatever way you want. The SDK does *not* gather screen media for you
+    const screenStream = await navigator.getDisplayMedia();
+
+    // add the media track(s) to the session
+    (session as ).addScreenTrackToSession()
+    sdk.acceptSession({ conversationId: session.conversationId, audioElement, videoElement });
```

Comment:
As I am thinking about this, would it make sense to have both options? 
1. `sdk.acceptSession({ mediaStream, ... })`
2. Keep `session.addScreenTrackToSession()` (or just keep `session.addTrack()` and document it)

Option 2 would allow for consumers to accept the session and then add media afterwards. That way they aren't waiting for the media before they are able to accept. But honestly, gathering screen media shouldn't take too long (unless they are using this in the browser which could take a long time).

Reply:
I was kind of on the fence about where to put the *add media* functions. I didn't want to use `session.addTrack` directly because I could see us needing to do extra stuff and if they use `session.addTrack` we'd either have to decorate that fn or make them use a different fn.

I don't mind adding support for `sdk.acceptSession` for mediaStream. I'll do that.

Reply:
Actually, I think I'm going to switch to the `acceptSession` route exclusively.